scripts/footprint/plot.py

```diff
@@ -0,0 +1,113 @@
+#!/usr/bin/env python3
+# Copyright (c) 2025 Basalte bv
+#
+# SPDX-License-Identifier: Apache-2.0
+
+"""
+A script to plot data in a sunburst chart generated by size_report.
+When you call the ram_report or rom_report targets you end up
+with a json file in the build directory that can be used as input
+for this script.
+
+Example:
+   ./scripts/footprint/plot.py build/ram.json
+
+Requires plotly to be installed, for example with pip:
+   pip install plotly
```

Comment:
Maybe add this to `scripts/requirements-extras.txt` so that most users pick this dependency up by default?

Reply:
I preferred to keep the script and its dependency as standalone. The error message should be clear enough IMO, but if others share your opinion, we could add it.

Reply:
Yes, it is pretty clear, and I'm not sure what the process of introducing a dependency is. Let's wait for others

Reply:
Update: added CMake targets too, and because of this added the dependency as you suggested.